Kernel/Devices/USB/USBDevice.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/OwnPtr.h>
+#include <AK/Types.h>
+#include <AK/Vector.h>
+#include <Kernel/Devices/USB/UHCIController.h>
+#include <Kernel/Devices/USB/USBDescriptors.h>
+#include <Kernel/Devices/USB/USBDevice.h>
+#include <Kernel/Devices/USB/USBRequest.h>
+
+static u32 s_next_usb_address = 1; // Next address we hand out to a device once it's plugged into the machine
+
+namespace Kernel::USB {
+
+void USBDevice::create(PortNumber port, DeviceSpeed speed)
+{
+    USBDevice device(port, speed);
+    device.enumerate();
+}
+
+USBDevice::USBDevice(PortNumber port, DeviceSpeed speed)
+    : m_device_port(port)
+    , m_device_speed(speed)
+    , m_address(0)
+{
+    // FIXME: How should we propagate the error from `create_pipe` here in the constructor???
```

Comment:
Please write comments in proper English (where questions end in a single `?`) :^)

Also try to avoid asking questions in comments if it's possible to formulate something actionable instead, for example:
`FIXME: Figure out how to propagate the error from create_pipe() to our caller.`

And since we're on the subject, the way we normally propagate errors is by making sure they happen *before* we're in a constructor. I suggest making a factory function like so:

```cpp
KResultOr<NonnullOwnPtr<USBDevice>> USBDevice::try_create(PortNumber port, DeviceSpeed speed)
{
    auto pipe_or_error = USBPipe::try_create_pipe(USBPipe::Type::Control, USBPipe::Direction::In, 0, 8, 0);
    if (pipe_or_error.is_error())
        return pipe_or_error.error();
    auto device = adopt_own_if_nonnull(new USBPipe(port, speed, pipe_or_error.release_value());
    if (!device)
        return ENOMEM;
    return device.release_nonnull();
}
```

Reply:
Cheers AK! I'll do that :^) [I'll try and reign in my super casual comment style hehe]